tests/v2/validation/fleet/provisioning/airgap/airgap.go

```diff
@@ -0,0 +1,193 @@
+	var key ssh.Signer
+	var err error
+
+	if strings.Contains(privateKey, rsa) {
+
+		keyBlock, _ := pem.Decode([]byte(privateKey))
+		if keyBlock == nil {
+			return "", errors.New("unable to use privateKey")
+		}
+
+		rsaKey, err := x509.ParsePKCS1PrivateKey(keyBlock.Bytes)
+		if err != nil {
+			return "", err
+		}
+
+		key, err = ssh.NewSignerFromKey(rsaKey)
+	} else {
+		key, err = ssh.ParsePrivateKey([]byte(privateKey))
+	}
+	if err != nil {
+		return "", err
+	}
+
+	return updateLocalFleetRepo(bastionUser, bastionIP, key)
+
+}
+
+// updateLocalFleetRepo clones fleet-examples repo onto the bastion node via ssh
+func updateLocalFleetRepo(bastionUser, bastionIP string, privateKey ssh.Signer) (string, error) {
+	config := &ssh.ClientConfig{
+		User: bastionUser,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(privateKey),
+		},
+		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	conn, err := ssh.Dial(tcp, bastionIP+sshPort, config)
+	if err != nil {
+		return "", err
+	}
+	defer conn.Close()
+
+	gitSession, err := conn.NewSession()
+	if err != nil {
+		return "", err
+	}
+	defer gitSession.Close()
+
+	var b strings.Builder
+	gitSession.Stdout = &b
+	gitSession.Stderr = &b
+
+	// install git if it isn't on the bastion already
+	err = gitSession.Run("git version")
+	if err != nil {
+		if strings.Contains(b.String(), "no package named") {
+			installGitSession, err := conn.NewSession()
+			if err != nil {
+				return "", err
+			}
+			defer installGitSession.Close()
+
+			osOutput, err := installGitSession.CombinedOutput("hostnamectl")
+			if err != nil {
+				return "", err
+			}
+
+			osString := string(osOutput)
+
+			if strings.Contains(osString, "Ubuntu") {
+				err = executeCommand(conn, "sudo apt-get update && sudo apt-get install -y git")
+			} else if strings.Contains(osString, "SUSE") {
+				err = executeCommand(conn, "sudo zypper refresh && sudo zypper install git -y")
+			}
+			if err != nil {
+				return "", err
+			}
+
+		} else {
+			return "", err
+		}
+	}
+
+	err = executeCommand(conn, "cd ~/ && git clone --mirror "+rancherGithubURL+fleetExampleFolderName)
+	if err != nil {
+		return "", err
+	}
+
+	err = executeCommand(conn, "sudo git config --system --add safe.directory '*'")
+	if err != nil {
+		return "", err
+	}
+
+	session, err := conn.NewSession()
+	if err != nil {
+		return "", err
+	}
+	defer session.Close()
+
+	internalByteIP, err := session.CombinedOutput("hostname -I | awk '{print $1}'")
+
+	// output ends in a newline, which we don't need
+	return string(internalByteIP)[:len(internalByteIP)-1], err
+}
+
+// executeCommand executes a command via ssh. Each executed command needs its own session. See ssh package for more details.
+func executeCommand(client *ssh.Client, command string) error {
+	session, err := client.NewSession()
+	if err != nil {
+		return err
+	}
+	defer session.Close()
+
+	session.Stdout = os.Stdout
+	session.Stderr = os.Stderr
```

Comment:
not a comment but more of a clarifying question, is this going to print the output in the terminal?

Reply:
during testing, I did not see it output to the terminal unless there's an error.